packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts

```diff
@@ -224,28 +226,38 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
       WHERE "objectMetadataId" = '${createdFieldMetadata.objectMetadataId}'`,
         );
 
-        const existingViewFields = await workspaceQueryRunner?.query(
-          `SELECT * FROM ${dataSourceMetadata.schema}."viewField"
+        if (!isEmpty(view)) {
+          const existingViewFields = (await workspaceQueryRunner?.query(
+            `SELECT * FROM ${dataSourceMetadata.schema}."viewField"
       WHERE "viewId" = '${view[0].id}'`,
-        );
+          )) as ViewFieldWorkspaceEntity[];
+
+          const createdFieldIsAlreadyInView = existingViewFields.some(
+            (existingViewField) =>
+              existingViewField.fieldMetadataId === createdFieldMetadata.id,
+          );
 
-        const lastPosition = existingViewFields
-          .map((viewField) => viewField.position)
-          .reduce((acc, position) => {
-            if (position > acc) {
-              return position;
-            }
+          if (!createdFieldIsAlreadyInView) {
```

Comment:
Added this as we have no constraint elsewhere to forbid adding an already present field in a view - not sure if there is a usecase where that would make sense?
on my side I needed this in case of a rollback where we would restore the initial field by creating it again which would trigger the field's addition to the view, while it was still present in the view (from when it was originally created as we don't delete fields from viewFields when we delete the field - not sure why we don't do it?), creating a duplicated presence in the view.

Reply:
AFAIK adding the same field twice is currently not possible on frontend side. I guess that means we do not have any use case. 

About your second question, I do we still see the viewFields when we delete le field currently? I don't get how this would work